src/core/agent/services/connectionService.ts

```diff
@@ -369,42 +367,36 @@ class ConnectionService extends AgentService {
   async resolveOobi(
     url: string,
     waitForCompletion = true
-  ): Promise<Operation & { response: State } & Contact> {
-    const alias = new URL(url).searchParams.get("name") ?? uuidv4();
-    const groupId = new URL(url).searchParams.get("groupId") ?? "";
-    const operation = (await waitAndGetDoneOp(
-      this.props.signifyClient,
-      await this.props.signifyClient.oobis().resolve(url, alias),
-      5000
-    )) as Operation & { response: State };
-
-    if (!operation.done && !waitForCompletion) {
+  ): Promise<{op: Operation, connection?: Contact, alias: string}> {
+    const alias = new URL(url).searchParams.get("name") ?? randomSalt();
+    let connection;
+    let operation: Operation & { response: State };
+    if(waitForCompletion){
+      operation = (await waitAndGetDoneOp(
+        this.props.signifyClient,
+        await this.props.signifyClient.oobis().resolve(url, alias),
+        5000
+      )) as Operation & { response: State };
+      if (!operation.done) {
+        throw new Error(ConnectionService.FAILED_TO_RESOLVE_OOBI);
+      }
+      if(operation.response && operation.response.i){
+        const connectionId = operation.response.i
+        connection = await this.props.signifyClient.contacts().update(connectionId, { alias });
+      }
+    }
+    else {
+      operation = await this.props.signifyClient.oobis().resolve(url, alias);
       const pendingOperation = await this.operationPendingStorage.save({
         id: operation.name,
         recordType: OperationPendingRecordType.Oobi,
       });
-
       this.props.eventEmitter.emit<OperationAddedEvent>({
         type: EventTypes.OperationAdded,
         payload: { operation: pendingOperation },
       });
-    } else if (!operation.done) {
-      throw new Error(ConnectionService.FAILED_TO_RESOLVE_OOBI);
     }
-
-    const connectionId =
-      operation.done && operation.response
-        ? operation.response.i
-        : new URL(url).pathname.split("/oobi/").pop()!.split("/")[0];
-
-    const keriaConnection = await this.props.signifyClient
-      .contacts()
-      .update(connectionId, {
-        alias,
-        groupCreationId: groupId,
-        createdAt: new Date().toISOString(),
-      });
-    const oobi = { ...operation, ...keriaConnection };
+    const oobi = { op: operation, connection, alias };
```

Comment:
Can we just call `return { op: operation, connection, alias };`? This object doesn't represent an OOBI, it's more than that.